src/components/viewer.ts

```diff
@@ -109,14 +72,14 @@ export class Viewer {
         return url
     }
 
-    openViewer(sourceFile: string) {
-        const url = this.checkViewer(sourceFile, 'viewer')
+    openBrowser(sourceFile: string, newViewer: boolean = false) {
+        const url = this.checkViewer(sourceFile, 'browser', newViewer = newViewer)
```

Comment:
You are missing the `respectOutDir` parameter before `newViewer` here. Without that change `newViewer` actually sets the `respectOutDir` parameter of the function, meaning that you can't open a viewer in the browser if you have an outDir specified.